File: backend/src/main/java/com/scmc/domain/dto/EncryptResponse.java
package com.scmc.domain.dto;

import com.scmc.domain.dto.audit.AuditStep;
import java.util.List;

public record EncryptResponse (

    String originalMessage,

    String paddedMessage,

    String permutedMessage,

    String encryptedMessage,

    Integer blockSize,

    List<Integer> permutation,

    Integer shift,

    List<AuditStep> audit
){

  }

File: frontend/SCMC_Frontend/src/modules/encrypt/types/EncryptResponse.ts
import type { AuditStep } from "./AuditStep";

export interface EncryptResponse{

    originalMessage:string;

    paddedMessage:string;

    permutedMessage:string;

    encryptedMessage:string;

    blockSize:number;

    permutation:number[];

    shift:number;

    audit:AuditStep[];

}